File: moveZeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int read = 0, write = 0;

        for (int i=0; i<nums.size(); i++){
            if (nums[read] == 0){
                read++;
            }
            else{
                int temp = nums[write];
                nums[write] = nums[read];
                nums[read] = temp;
                write++;
                read++;
            }
        }
    }
};

File: moveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i,count = 0,0
        while count < len(nums):
            if nums[i] == 0:
                nums.remove(nums[i])
                nums.append(0)
            else:
                i+=1
            count+=1
        
